packages/shared/src/cachedFetch.ts

```diff
@@ -66,3 +110,11 @@ export function resetCache(url?: string) {
     cache.clear();
   }
 }
+
+function storeCachedEntry(url: string, entry: CacheEntry) {
+  const deferred = cache.get(url)?.deferred;
+  assert(deferred, "Deferred should be defined");
```

Comment:
nit: The caller could pass in the `Deferred` instance (it has it already) then you could avoid this check.